src/middlewared/middlewared/plugins/pool_/import_pool.py

```diff
@@ -227,6 +232,7 @@ def import_on_boot_impl(self, vol_name, vol_guid, set_cachefile=False):
             self.logger.error('Unhandled exception importing %r', vol_name, exc_info=True)
             return False
 
+        self.middleware.call_sync('pool.handle_unencrypted_datasets_on_import', vol_name)
```

Comment:
Why would you call this here and not in `unlock_on_boot_impl`? We're making 2 duplicate calls to `pool.dataset.get_instance_quick`